plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/CorrelationParamsContext.java

```diff
@@ -0,0 +1,166 @@
+ * Defines vector similarity function, m and ef_construction hyper parameters field mappings for correlation_vector type.
+ *
+ * @opensearch.internal
+ */
+public class CorrelationParamsContext implements ToXContentFragment, Writeable {
+
+    /**
+     * Vector Similarity Function field
+     */
+    public static final String VECTOR_SIMILARITY_FUNCTION = "similarityFunction";
+    /**
+     * Parameters field to define m and ef_construction
+     */
+    public static final String PARAMETERS = "parameters";
+
+    private final VectorSimilarityFunction similarityFunction;
+    private final Map<String, Object> parameters;
+
+    /**
+     * Parameterized ctor for CorrelationParamsContext
+     * @param similarityFunction Vector Similarity Function
+     * @param parameters Parameters to define m and ef_construction
+     */
+    public CorrelationParamsContext(VectorSimilarityFunction similarityFunction, Map<String, Object> parameters) {
+        this.similarityFunction = similarityFunction;
+        this.parameters = parameters;
+    }
+
+    /**
+     * Parameterized ctor for CorrelationParamsContext
+     * @param sin StreamInput
+     * @throws IOException IOException
+     */
+    public CorrelationParamsContext(StreamInput sin) throws IOException {
+        this.similarityFunction = VectorSimilarityFunction.valueOf(sin.readString());
+        if (sin.available() > 0) {
+            this.parameters = sin.readMap();
+        } else {
+            this.parameters = null;
+        }
+    }
+
+    /**
+     * Parse into CorrelationParamsContext
+     * @param in Object
+     * @return CorrelationParamsContext
+     */
+    public static CorrelationParamsContext parse(Object in) {
+        if (!(in instanceof Map<?, ?>)) {
+            throw new MapperParsingException("Unable to parse CorrelationParamsContext");
+        }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> contextMap = (Map<String, Object>) in;
+        VectorSimilarityFunction similarityFunction = VectorSimilarityFunction.EUCLIDEAN;
+        Map<String, Object> parameters = new HashMap<>();
+
+        if (contextMap.containsKey(VECTOR_SIMILARITY_FUNCTION)) {
+            Object value = contextMap.get(VECTOR_SIMILARITY_FUNCTION);
+
+            if (value != null && !(value instanceof String)) {
+                throw new MapperParsingException(String.format(Locale.getDefault(), "%s must be a string", VECTOR_SIMILARITY_FUNCTION));
+            }
+
+            try {
+                similarityFunction = VectorSimilarityFunction.valueOf((String) value);
+            } catch (IllegalArgumentException ex) {
+                throw new MapperParsingException(String.format(Locale.getDefault(), "Invalid %s: %s", VECTOR_SIMILARITY_FUNCTION, value));
+            }
+        }
+        if (contextMap.containsKey(PARAMETERS)) {
+            Object value = contextMap.get(PARAMETERS);
+            if (value == null) {
+                parameters = null;
+            }
```

Comment:
Why do this? The next line asserts `value` is a `Map` or else throws. Am I missing something here?

Reply:
removed the line.